core/src/ipc.rs

```diff
@@ -134,3 +140,107 @@ impl<T> FromRawFd for IpcReceiver<T> {
         }
     }
 }
+
+/// Creates a pair of sockets that are meant for passing file descriptors to
+/// partitions.
+pub fn io_pair<T>() -> TypedResult<(IoSender<T>, IoReceiver<T>)> {
+    let (tx, rx) = socketpair(
+        AddressFamily::Unix,
+        SockType::Datagram,
+        None,
+        SockFlag::empty(),
+    )
+    .typ(SystemError::Panic)?;
+    unsafe { Ok((IoSender::from_raw_fd(tx), IoReceiver::from_raw_fd(rx))) }
+}
+
+#[derive(Debug)]
+/// Internal data type for the IO resource sender
+pub struct IoSender<T> {
+    socket: UnixDatagram,
+    _p: PhantomData<T>,
+}
+
+impl<T> IoSender<T>
+where
+    T: AsRawFd,
+{
+    /// Sends a resource to the receiving socket.
+    pub fn try_send(&self, resource: UdpSocket) -> TypedResult<()> {
+        let fds = [resource.as_raw_fd()];
+        let cmsg = [ControlMessage::ScmRights(&fds)];
+        let buffer = [0u8; 1];
```

Comment:
should we add an identifier to the config for each port and use this buffer to identify it?